src/main/java/org/jabref/model/entry/BibEntrySerializer.java

```diff
@@ -0,0 +1,42 @@
+package org.jabref.model.entry;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.InternalField;
+
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class BibEntrySerializer implements JsonSerializer<BibEntry> {
+    @Override
+    public JsonElement serialize(BibEntry entry, Type typeOfSrc, JsonSerializationContext context) {
+        JsonObject obj = new JsonObject();
+
+        obj.addProperty("type", entry.getType().getName());
```

Comment:
Please introduce a constant (`public static final String JSON_TYPE`)

General thing: We need to add two meta information to the JSON. We need to the bibtex content and the meta data type and key. They should not be mixed up. We need to consider that a user can add custom fields.

To be prepared for moree meta data, I would suggest a nested object

```json
{
  "bibtex_metadata": {
    "type": "article",
    "key": "the-citation-key"
  }
}
```

In case a user adds `bibtex_metadata` as custom field in the entry editor, then the thing breaks.

Alternative: Do not add special handling for  `citationkey`, because JabRef itself uses that field name.

Another thing:

Shouldn't one use `typedEntry.getTypeForDisplay()` for type writing (Source: org.jabref.logic.bibtex.BibEntryWriter#writeRequiredFieldsFirstRemainingFieldsSecond)

```java
  TypedBibEntry typedEntry = new TypedBibEntry(entry, bibDatabaseMode);
  typedEntry.getTypeForDisplay();
```